pkg/helm/factory.go

```diff
@@ -0,0 +1,126 @@
+package helm
+
+import (
+	"context"
+	"reflect"
+	"sync"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	configcli "github.com/aws/eks-anywhere/pkg/config"
+	"github.com/aws/eks-anywhere/pkg/controller/clusters"
+	"github.com/aws/eks-anywhere/pkg/registrymirror"
+)
+
+// Config contains configuration options for Helm.
+type Config struct {
+	RegistryMirror *registrymirror.RegistryMirror
+	ProxyConfig    map[string]string
+	Insecure       bool
+}
+
+// Opt is a functional option for configuring Helm behavior.
+type Opt func(*Config)
+
+// Client represents a Helm client.
+type Client interface {
```

Comment:
nit: can we make this file smaller by moving stuff out?
we have polluted a bit the responsability
maybe have a client.go and a config.go